tools/pubsys/src/aws/ami/register/mod.rs

```diff
@@ -48,6 +50,16 @@ async fn _register_image(
     info!("Making register image call in {region}");
     let register_response = amispec
         .as_register_image_call()
+        .set_tag_specifications(tags.map(|x| {
+            vec![TagSpecification::builder()
+                .resource_type(ResourceType::Image)
+                .set_tags(Some(
+                    x.iter()
+                        .map(|(key, value)| Tag::builder().key(key).value(value).build())
+                        .collect(),
+                ))
+                .build()]
+        }))
         .send_with(ec2_client)
```

Comment:
This tag specification setting seems like it should be done as part of `as_register_image_call()` rather than separately here at the call site.